home/screens/AccountModal/LoggedOutAccountView.tsx

```diff
@@ -8,17 +9,40 @@ import url from 'url';
 import Analytics from '../../api/Analytics';
 import ApolloClient from '../../api/ApolloClient';
 import Config from '../../api/Config';
-import { useDispatch } from '../../redux/Hooks';
+import { useDispatch, useSelector } from '../../redux/Hooks';
 import SessionActions from '../../redux/SessionActions';
 import { useAccountName } from '../../utils/AccountNameContext';
+import hasSessionSecret from '../../utils/hasSessionSecret';
 
-export function LoggedOutAccountView() {
+type Props = {
+  refetch: () => Promise<void>;
+};
+
+export function LoggedOutAccountView({ refetch }: Props) {
   const dispatch = useDispatch();
   const [isAuthenticating, setIsAuthenticating] = React.useState(false);
+  const [isFinishedAuthenticating, setIsFinishedAuthenticating] = React.useState(false);
   const [authenticationError, setAuthenticationError] = React.useState<string | null>(null);
   const mounted = React.useRef<boolean | null>(true);
   const theme = useExpoTheme();
   const { setAccountName } = useAccountName();
+  const navigation = useNavigation();
+
+  const { sessionSecretExists } = useSelector((data) => {
+    const sessionSecretExists = hasSessionSecret(data.session);
+    return {
+      sessionSecretExists,
+    };
+  });
+
+  React.useEffect(() => {
+    // after logging in, wait for redux action to dispatch, refetch with new sessionSecret, then dismiss modal
+    if (isFinishedAuthenticating && sessionSecretExists) {
+      refetch().then(() => {
```

Comment:
what happens in the case that refetch rejects?

also, we could move this into an async function rather than using promise directly. this is a common convention that i personally prefer

```js
React.useEffect(() => {
  async function refetchThenGoBackAsync() {
    try {
      if (isFinishedAuthenticating && sessionSecretExists) {
        await refetch();
        navigation.goBack();
      }
    } catch (e) {
      // handle error
    }
  }

  refetchThenGoBackAsync();
});
```

Reply:
I acknowledge this case in the comment I added to this code: https://github.com/expo/expo/pull/17233/commits/366f5a8e4e67071ce21c207ee043763eedcacd61#diff-642253f7852d157e1e2b30b5f2b10dfd8e94a0d482741e255bb5ba08049b17d9R45